cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,98 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
+import SignInPageObject from '../support/pages/signIn.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import SettingsPageObject from '../support/pages/settings.pageObject';
+import faker from 'faker';
+import {
+  updateSuccessMessage,
+  confirmationMessage
+} from '../plugins/alertMessages';
+
+const signInPage = new SignInPageObject();
+const homePage = new HomePageObject();
+const settingsPage = new SettingsPageObject();
 
 describe('Settings page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+      cy.registerUser(user.email, user.username, user.password);
+    });
   });
 
   beforeEach(() => {
+    signInPage.visit();
+  });
 
+  afterEach(() => {
+    settingsPage.logOutSettings();
```

Comment:
Avoid frequent logging out in 'afterEach'. This could make tests run slower. Instead consider clearing user session data or using a new user for each test.

Reply:
You wrote earlier:
Logging out should be in afterEach hook to make sure each test is independent and to improve code readability.
[https://github.com/mate-academy/qa_cypress_e2e_project/pull/59#discussion_r1291103455](url)